dbms/src/Functions/tests/gtest_duration_extract.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Columns/ColumnConst.h>
+#include <Columns/ColumnString.h>
+#include <Common/Exception.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionsDuration.h>
+#include <Interpreters/Context.h>
+#include <TestUtils/FunctionTestUtils.h>
+#include <TestUtils/TiFlashTestBasic.h>
+
+#include <string>
+#include <vector>
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wsign-compare"
+#include <Poco/Types.h>
+
+#pragma GCC diagnostic pop
+
+namespace DB
+{
+namespace tests
+{
+// TODO: rewrite using executeFunction()
+class TestDurationExtract : public DB::tests::FunctionTest
+{
+};
+
+TEST_F(TestDurationExtract, ExtractFromMyDuration)
+try
+{
+    const Context context = TiFlashTestEnv::getContext();
```

Comment:
1. There are some test utils to wrap the details of creating function/creating block, it is not recommanded to write function test using the low level method. Please refer to `gtet_date_add` for details.
2. Please add more tests to test if the second argument is constant/nullable vector